python/lsst/pipe/tasks/makeDirectWarp.py

```diff
@@ -0,0 +1,799 @@
+
+        Specifically, this method updates the following (if available) to the
+        input exposure ``exp`` in place from ``visit_summary``:
+
+        - Aperture correction map
+        - Photometric calibration
+        - PSF
+        - WCS
+
+        Parameters
+        ----------
+        exp : `~lsst.afw.image.Exposure`
+            Exposure to be updated.
+        visit_summary : `~lsst.afw.table.ExposureCatalog` | None
+            Table of visit summary information.  If not None, the visit summary
+            information will be used to update the calibration of the input
+            exposures. Otherwise, the input exposures will be used as-is.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if ``visit_summary`` is provided but does not contain a
+            record corresponding to ``exp``.
+        """
+        if not visit_summary:
+            logger.debug("No visit summary provided.")
+            # Question: Should we not apply background and skyCorr???
+        else:
+            logger.debug("Updating calibration from visit summary.")
+
+        if background:
+            exp.maskedImage += background.getImage()
+
+        if visit_summary:
+            detector = exp.info.getDetector().getId()
+            row = visit_summary.find(detector)
+
+            if row is None:
+                raise RuntimeError(
+                    f"Unexpectedly incomplete visit_summary: {detector=} is missing."
+                )
+
+            if photo_calib := row.getPhotoCalib():
+                exp.setPhotoCalib(photo_calib)
+            else:
+                logger.warning(
+                    "No photometric calibration found in visit summary for detector = %s.",
+                    detector,
+                )
+
+            if wcs := row.getWcs():
+                exp.setWcs(wcs)
+            else:
+                logger.warning(
+                    "No WCS found in visit summary for detector = %s.", detector
+                )
+
+            if psf := row.getPsf():
+                exp.setPsf(psf)
+            else:
+                logger.warning(
+                    "No PSF found in visit summary for detector = %s.", detector
+                )
+
+            if apcorr_map := row.getApCorrMap():
+                exp.setApCorrMap(apcorr_map)
+            else:
+                logger.warning(
+                    "No aperture correction map found in visit summary for detector = %s.",
+                    detector,
+                )
+
+        # Calibrate the (masked) image.
+        # This should likely happen even if visit_summary is None.
+        photo_calib = exp.getPhotoCalib()
+        exp.maskedImage = photo_calib.calibrateImage(
+            exp.maskedImage, includeScaleUncertainty=includeScaleUncertainty
+        )
+        exp.maskedImage /= photo_calib.getCalibrationMean()
+
+        if skyCorr:
+            exp.maskedImage -= skyCorr.getImage()
```

Comment:
If the inputs have absolute calibration, would the order matter (since the scaling would be by 1.0 I believe). I can bring it up on Slack if you're unsure.

Reply:
Ah, if this works because we're dividing by 1.0, that would explain why it hasn't appeared as a problem already.  But I think it does mean this code is still fragile to `getCalibrationMean()` not returning `1.0`.

Reply:
Writing here for posterity. After discussions on [Slack](https://lsstc.slack.com/archives/C2JPMCF5X/p1711463918984589) and during our pair coding, we saw that the calexp is unchanged modulo floating point rounding errors and decided to move the skyCorr correction prior to calibrating the image.